core/src/main/java/org/elasticsearch/search/aggregations/support/InternalOrder.java

```diff
@@ -327,40 +342,82 @@ public static void writeOrder(Terms.Order order, StreamOutput out) throws IOExce
                 out.writeString(path.toString());
             } else if (order instanceof CompoundOrder) {
                 CompoundOrder compoundOrder = (CompoundOrder) order;
-                    out.writeByte(order.id());
-                    out.writeVInt(compoundOrder.orderElements.size());
-                    for (Terms.Order innerOrder : compoundOrder.orderElements) {
-                        Streams.writeOrder(innerOrder, out);
-                    }
+                out.writeByte(order.id());
+                out.writeVInt(compoundOrder.orderElements.size());
+                for (BucketOrder innerOrder : compoundOrder.orderElements) {
+                    Streams.writeOrder(innerOrder, out);
+                }
             } else {
                 out.writeByte(order.id());
             }
         }
 
-        public static Terms.Order readOrder(StreamInput in) throws IOException {
-            return readOrder(in, false);
-        }
-
-        public static Terms.Order readOrder(StreamInput in, boolean absoluteOrder) throws IOException {
+        public static BucketOrder readOrder(StreamInput in) throws IOException {
+            // TODO backwards compat
```

Comment:
What is the expected behavior for the `writeTo` method for the `histogram` if the `order` is a `CompundOrder`? I think the best option is just to serialize the first order? However wont this produce unexpected results if the cluster has mixed versions? I'm not sure how the transport client works in this case.